### ISSUE TITLE
Acceleration NotImplemented error with third party S3 implementation

### ISSUE THREAD

**Issue by dr4Ke** [hashicorp/terraform-provider-aws#13726]
When using a different S3 implementation (here Cloudian), the bucket gets created but the terraform states still fails, due to Acceleration not being implemented server side.

### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or other comments that do not add relevant new information or questions, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Terraform Version

```
Terraform v0.12.24
+ provider.aws v2.66.0
```

### Affected Resource(s)

* aws_s3_bucket

### Terraform Configuration Files

```hcl
provider "aws" {
  endpoints {
    s3 = "https://object-fr.antemeta.net"
  }
  region = "fr"
  skip_region_validation = true
  skip_get_ec2_platforms = true
  s3_force_path_style = true
  skip_credentials_validation = true
  skip_metadata_api_check     = true
  skip_requesting_account_id  = true
}

resource "aws_s3_bucket" "test-bucket" {
  bucket = "test-bucket"
  acl    = "private"
  acceleration_status = null
}
```

### Debug Output

https://gist.github.com/dr4Ke/b27bdf40fa54dbe9da92a9085380cdeb

### Panic Output

N/A

### Expected Behavior

The bucket gets created and `terraform apply` succeeds.

### Actual Behavior

The bucket gets created but the `terraform apply` fails with error:
```
Error: error getting S3 Bucket acceleration configuration: NotImplemented: A header you provided implies functionality that is not implemented.
	status code: 501, request id: , host id:
```

It is due to the Acceleration not being implemented. Subsequents run of `terraform plan` gives the same error.

Detailed output:
```
2020-06-12T08:58:16.839Z [DEBUG] plugin.terraform-provider-aws_v2.66.0_x4: -----------------------------------------------------
2020-06-12T08:58:16.839Z [DEBUG] plugin.terraform-provider-aws_v2.66.0_x4: 2020/06/12 08:58:16 [DEBUG] [aws-sdk-go] <?xml version="1.0" encoding="UTF-8"?><VersioningConfiguration xmlns="http://s3.amazonaws.com/doc/2006-03-01/"></VersioningConfiguration>
2020-06-12T08:58:16.839Z [DEBUG] plugin.terraform-provider-aws_v2.66.0_x4: 2020/06/12 08:58:16 [DEBUG] Waiting for state to become: [success]
2020-06-12T08:58:16.839Z [DEBUG] plugin.terraform-provider-aws_v2.66.0_x4: 2020/06/12 08:58:16 [DEBUG] [aws-sdk-go] DEBUG: Request s3/GetBucketAccelerateConfiguration Details:
2020-06-12T08:58:16.839Z [DEBUG] plugin.terraform-provider-aws_v2.66.0_x4: ---[ REQUEST POST-SIGN ]-----------------------------
2020-06-12T08:58:16.839Z [DEBUG] plugin.terraform-provider-aws_v2.66.0_x4: GET /test-bucket?accelerate= HTTP/1.1
2020-06-12T08:58:16.839Z [DEBUG] plugin.terraform-provider-aws_v2.66.0_x4: Host: object-fr.antemeta.net
2020-06-12T08:58:16.839Z [DEBUG] plugin.terraform-provider-aws_v2.66.0_x4: User-Agent: aws-sdk-go/1.31.2 (go1.13.7; linux; amd64) APN/1.0 HashiCorp/1.0 Terraform/0.12.24 (+https://www.terraform.io)
2020-06-12T08:58:16.839Z [DEBUG] plugin.terraform-provider-aws_v2.66.0_x4: Authorization: AWS4-HMAC-SHA256 Credential=XXXXXXXXXXXXXXX/20200612/fr/s3/aws4_request, SignedHeaders=host;x-amz-content-sha256;x-amz-date, Signature=791ca9bfd2fee30397381f6c31463ddb9497f255b336e2ae5227b50a44665d22
2020-06-12T08:58:16.839Z [DEBUG] plugin.terraform-provider-aws_v2.66.0_x4: X-Amz-Content-Sha256: e3b0c44298fc1c149afbf4c8996fb92427ae41e4649b934ca495991b7852b855
2020-06-12T08:58:16.839Z [DEBUG] plugin.terraform-provider-aws_v2.66.0_x4: X-Amz-Date: 20200612T085816Z
2020-06-12T08:58:16.839Z [DEBUG] plugin.terraform-provider-aws_v2.66.0_x4: Accept-Encoding: gzip
2020-06-12T08:58:16.839Z [DEBUG] plugin.terraform-provider-aws_v2.66.0_x4: 
2020-06-12T08:58:16.839Z [DEBUG] plugin.terraform-provider-aws_v2.66.0_x4: 
2020-06-12T08:58:16.839Z [DEBUG] plugin.terraform-provider-aws_v2.66.0_x4: -----------------------------------------------------
2020-06-12T08:58:17.053Z [DEBUG] plugin.terraform-provider-aws_v2.66.0_x4: 2020/06/12 08:58:17 [DEBUG] [aws-sdk-go] DEBUG: Response s3/GetBucketAccelerateConfiguration Details:
2020-06-12T08:58:17.053Z [DEBUG] plugin.terraform-provider-aws_v2.66.0_x4: ---[ RESPONSE ]--------------------------------------
2020-06-12T08:58:17.053Z [DEBUG] plugin.terraform-provider-aws_v2.66.0_x4: HTTP/1.1 501 Not Implemented
2020-06-12T08:58:17.053Z [DEBUG] plugin.terraform-provider-aws_v2.66.0_x4: Connection: close
2020-06-12T08:58:17.053Z [DEBUG] plugin.terraform-provider-aws_v2.66.0_x4: Content-Length: 286
2020-06-12T08:58:17.053Z [DEBUG] plugin.terraform-provider-aws_v2.66.0_x4: Content-Type: application/xml
2020-06-12T08:58:17.053Z [DEBUG] plugin.terraform-provider-aws_v2.66.0_x4: Date: Fri, 12 Jun 2020 08:58:17 GMT
2020-06-12T08:58:17.053Z [DEBUG] plugin.terraform-provider-aws_v2.66.0_x4: Server: CloudianS3
2020-06-12T08:58:17.053Z [DEBUG] plugin.terraform-provider-aws_v2.66.0_x4: Strict-Transport-Security: max-age=31536000
2020-06-12T08:58:17.053Z [DEBUG] plugin.terraform-provider-aws_v2.66.0_x4: X-Gmt-Error-Code: NotImplemented
2020-06-12T08:58:17.053Z [DEBUG] plugin.terraform-provider-aws_v2.66.0_x4: X-Gmt-Message: A header you provided implies functionality that is not implemented.
2020-06-12T08:58:17.053Z [DEBUG] plugin.terraform-provider-aws_v2.66.0_x4: 
2020-06-12T08:58:17.053Z [DEBUG] plugin.terraform-provider-aws_v2.66.0_x4: 
2020-06-12T08:58:17.053Z [DEBUG] plugin.terraform-provider-aws_v2.66.0_x4: -----------------------------------------------------

```

### Steps to Reproduce

1. `terraform apply`

### Important Factoids

Please note this is happening on a S3 implementation that's not Amazon's.

### References

* Similar to #11556 but does not get fixed by using `acceleration_status = null` parameter
* See [comment](https://github.com/terraform-providers/terraform-provider-aws/pull/12049#issuecomment-636225241) by @gdavison

**Comment by dr4Ke** [comment]
I didn't see a `skip_*` parameter for disabling acceleration management.

**Comment by hongkongkiwi** [comment]
A skip parameter would be perfect. I could also do with this for Wasabi S3 since it doesn't support replication.

**Comment by mflorin** [comment]
@dr4Ke @hongkongkiwi - I've opened up a pull request with the skip_* features implemented at https://github.com/hashicorp/terraform-provider-aws/pull/17564 - hopefully it goes through :) - you can have a look, see if it 
solves the problem for you as well.

**Comment by jerome-jutteau** [comment]
@mflorin Thanks for your PR, we are looking for similar change with even [more options](https://github.com/outscale-dev/terraform-provider-aws/commits/oos). Hope your PR will be approved.

**Comment by dabo-devconsole** [comment]
There are also providers, which do not support versioning e.g. Strato HiDrive S3.
It would be great to be able to also skip_versioning.

**Comment by mneverov** [comment]
According to the [FAQ](https://github.com/hashicorp/terraform-provider-aws/blob/514f96ca5aca7646c7e245b9eb08f8efd9e15744/docs/contributing/faq.md#how-do-you-decide-what-gets-merged-for-each-release) I can speed up the PR#17564 merge by describing bad UX. 
It is quite annoying not being able to use `s3_bucket` and having to craft s3 bucket manually with the `null_resource` and multiple bash commands:
```
    command = <<EOF
aws s3api create-bucket...
aws s3api put-bucket-lifecycle ...
aws s3api put-bucket-versioning ...
aws s3api put-bucket-encryption ...
EOF
```
It is slower, multiple commands instead of one increases possibility the bucket creation to fail, hard to debug what went wrong.
